[AlagappanRa] iP

AlagappanRa opened the pull request:
# DukePro
## Usage
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
   
DukePro is a **desktop app for managing tasks, optimized for use via a Command Line Interface** (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, DukePro can get your task management done faster than traditional GUI apps. 
It's
- **text-based**
- **easy to learn**
- ~~FAST~~ SUPER FAST to use

## Quick start

### Running the JAR file

1. Download the JAR file from [here]
2. Open the terminal and navigate to the directory where the JAR file is located
3. Run the JAR file using the command `java -jar ip.jar`

### Start using DukePro
+ Add a task using the `todo`, `deadline` or `event` command 
+ List all tasks using the `list` command 
+ Mark a task as done using the `done` command 
+ Delete a task using the `delete` command 
+ Find a task using the `find` command 
+ Exit the program using the `bye` command

### Features
- [x] Managing Tasks
- [x] Managing Deadlines
- [x] Managing Events
- [ ] Interactive GUI

### Special Features
+ What's more! Your tasks are **auto-saved** after every command.
No more worries about forgetting to save your tasks or if your 
computer crashes 😭😭😭.
__________________________________________________________________________________________

If you are a Java programmer, you can use it to practice Java too! Here's the `main` method:
```java
public class Duke {
    public static void main(String[] args) {
        new Duke("./data", "./data/store.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

AlagappanRa committed:
Tasks Completed:
Level 0. Rename, Greet, Exit
Level 1. Echo

AlagappanRa committed:
Tasks Completed:
Level 0. Rename, Greet, Exit
Level 1. Echo

AlagappanRa committed:
Level 3 [COMPLETED]

AlagappanRa committed:
Level 4 [COMPLETED]

AlagappanRa committed:
Level 5 [COMPLETED]

AlagappanRa committed:
Level 5 [COMPLETED]

AlagappanRa committed:
Level 6 [COMPLETED]

AlagappanRa committed:
Reformatted code to include additional enums

Reformatted code to include additional enums

AlagappanRa committed:
Level-7

AlagappanRa committed:
Merge pull request #1 from AlagappanRa/branch-Level-7

Level-7

AlagappanRa committed:
Level-8

AlagappanRa committed:
Merge pull request #2 from AlagappanRa/branch-Level-8

Level-8

AlagappanRa committed:
A-MoreOOP & A-Packages

AlagappanRa committed:
Merge pull request #3 from AlagappanRa/branch-Level-8

A-MoreOOP & A-Packages

AlagappanRa committed:
Merge pull request #4 from AlagappanRa/add-gradle-support

Add gradle support

AlagappanRa committed:
A-Gradle & A-JUnit

AlagappanRa committed:
A-JAR

AlagappanRa committed:
A-JavaDoc

AlagappanRa committed:
Merge pull request #5 from AlagappanRa/branch-Level-9

A-JavaDoc

billieboy7 commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,31 @@
+package commands;
```
will separating the "package commands;" and the rest of the imports according to their package help in improving user readability? For your consideration :)

billieboy7 commented on src/main/java/components/Storage.java:
```diff
@@ -0,0 +1,246 @@
+                    result.addTask(new Event(
+                        line.substring(7, fromIndex), 
+                        line.substring(fromIndex + 8, toIndex),
+                        line.substring(toIndex + 5, line.length() - 1)),
```
I like what you did here to improve the readability of your code as the readability is affected by the multiple nesting. Thanks for the inspiration!

billieboy7 commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,59 @@
+
+  @Override
+    public String toString() {
+        return String.format("[D]%s (by: %s)", super.toString(), date != null ?
```
Perhaps keeping the conditional operator in the same line could improve the readability of the code - easier to understand the logic.

billieboy7 commented on src/main/java/main/Duke.java:
```diff
@@ -0,0 +1,57 @@
+package main;
```
Your code in general is readable and clear! It is easy to follow. Great job on it!

AlagappanRa committed:
Add A-Coding Standard

AlagappanRa committed:
Merge branch 'master' into branch-A-CodingStandard

AlagappanRa committed:
Merge pull request #6 from AlagappanRa/branch-A-CodingStandard

Add A-Coding Standard

AlagappanRa committed:
Add find functionality

AlagappanRa committed:
Merge pull request #7 from AlagappanRa/branch-Level-9

Add find functionality

AlagappanRa committed:
Update README.md

ChangruHenryQian commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,32 @@
+import components.Ui;
+import tasks.TaskList;
+
+public abstract class Command {
```
Great work on creating a Command class and using it as a super class for all the commands supported.

ChangruHenryQian commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+        System.out.println("Got it. I've added this task:");
+        this.add(task);
+        System.out.println(" " + task);
+        System.out.println("Now you have " + Integer.toString(this.size()) + " " + (this.size() == 1 ? "task" : "tasks") + " in the list.");
```
I like your implementation on deciding the form of the noun for printing.

ChangruHenryQian commented on src/test/java/components/ParserTest.java:
```diff
@@ -0,0 +1,72 @@
+import commands.*;
+import org.junit.jupiter.api.Test;
+
+public class ParserTest {
```
Sufficient amount of test cases for the Parser class.

ChangruHenryQian commented on src/main/java/components/Parser.java:
```diff
@@ -0,0 +1,89 @@
+     * @return Deadline task.
+     * @throws DukeException If command is invalid.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```
Clean and neat implementation of the parse function.

ChangruHenryQian commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+            throw new DukeException("There is no task at that index.");
+        }
+
+        System.out.println(Ui.LINE);
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(" " + this.get(index - 1));
+        System.out.println(Ui.LINE);
```
Should this printing part be done by Ui to follow OOP?

ChangruHenryQian reviewed (commented):
Good job!

AlagappanRa committed:
Complete A-CheckStyle

AlagappanRa committed:
Update GUI

A CLI is being used.

GUI offers a more pleasurable user experience.

Lets refactor the code and add GUI to allow
for both a GUI and CLI option.

Keeping the CLI while improving the GUI allows
for extendibility of the code in the future.

AlagappanRa committed:
Update GUI

A CLI is being used.

GUI offers a more pleasurable user experience.

Lets refactor the code and add GUI to allow for both a GUI and CLI option.

Keeping the CLI while improving the GUI allows for extendibility of the code in the future.

AlagappanRa committed:
Merge pull request #8 from AlagappanRa/branch-Level-10

Update GUI

AlagappanRa committed:
Add assertions

Code lacks assertions.

Without assertions, we will not be able to test out code.

Lets add assertions within all abstract classes and to
all classes under components package.

Assertions are only added to abstract classes so that
the assertions do not have to be repeated for child
subclasses.

AlagappanRa committed:
Add assertions

Code lacks assertions.

Without assertions, we will not be able to test out code.

Lets add assertions within all abstract classes and to all classes under components package.

Assertions are only added to abstract classes so that the assertions do not have to be repeated for child subclasses.

AlagappanRa committed:
Update TaskList

TaskList does not use streams.

Streams provide a declarative way to write
code, making out code more understandable.

The tasklist :: printList() method is changed
to create a stream.

printList() is the most relevant method to change
as the collectors to list method is natural with
the print function together with the "\n".

AlagappanRa committed:
Add duplicate task handling

Duplicate tasks is addable.

We do not want duplicate tasks to be addable by default.

Lets make the the application reply to the user with a message
saying they tried to add a duplicate description upon
attempting to add duplicate description.

Allows the user to know if they have accidentally added a
overlapping task description.

AlagappanRa committed:
Fix bugs

Bugs are present.

We do not want bugs in the repository.

Lets make the bugs cause the least intrusion
to user experience as possible.

Removing bugs allows unexpected behaviours
to be removed.

AlagappanRa committed:
Last commit

AlagappanRa committed:
Resolved merge conflicts in favor of master

AlagappanRa committed:
Duplicate Commit

Auto grader script did not recognise
this branch.

AlagappanRa committed:
Merge pull request #13 from AlagappanRa/branch-A-CodeQuality

Branch a code quality

AlagappanRa committed:
Merge branch 'master' into branch-Level-10

AlagappanRa committed:
Merge pull request #14 from AlagappanRa/branch-Level-10

Merge branch-Level-10 into master

AlagappanRa committed:
Commit to restart failed deploy process.

AlagappanRa committed:
Yet another commit
